controllers/openshift_controller.go

```diff
@@ -89,161 +89,30 @@ func (r *KataConfigOpenShiftReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 		if !oldest && err != nil {
 			return reconcile.Result{Requeue: true}, err
 		} else if !oldest && err == nil {
+			r.Log.Info("not oldest CR")
```

Comment:
We only allow one instance of the kataconfig CR (what would be the point of two?). But this requirement is rather unusual because with most other application level operators you can have many, think of a database or something. 

"not oldest CR" means this reconcile loop is operating on a second CR. We consider the kataconfig that was created first to be the legit one. So any CR that is not the oldest would be not supported and we write a log to announce that and also set an error in the kataconfig status description.

I think Harshal made an attempt to solve it differently but it wasn't possible at the time. We can revisit for 4.9 if there's another way like a webhook or a singleton operator feature that was discussed in the past.

Reply:
Ack.  Feel free to open an issue to track this one, if you think that's needed.  I'm fine with the code as it's now, it was just not so easy to understand without the background.

Thanks for the explanation.